cli/flox/src/commands/show.rs

```diff
@@ -158,3 +173,40 @@ fn render_show(search_results: &[SearchResult], all: bool) -> Result<()> {
     println!("    {pkg_name} - {versions}");
     Ok(())
 }
+
+#[cfg(test)]
+mod test {
+    use catalog_api_v1::types::{self as api_types};
+    use flox_rust_sdk::flox::test_helpers::flox_instance;
+    use flox_rust_sdk::providers::catalog::Client;
+
+    use super::*;
+
+    #[tokio::test]
+    async fn show_handles_404() {
+        let (mut flox, _temp_dir_handle) = flox_instance();
+        let Client::Mock(ref mut client) = flox.catalog_client.as_mut().unwrap() else {
+            panic!()
+        };
+        client.push_error_response(
+            api_types::ErrorResponse {
+                detail: "detail".to_string(),
+                status_code: 404,
+            },
+            404,
+        );
+        let search_term = "search_term";
+        let err = Show {
+            all: false,
+            search_term: search_term.to_string(),
+        }
+        .handle(flox)
+        .await
+        .unwrap_err();
+
+        assert_eq!(
```

Comment:
As per above, I think a 404 on `/search` should be an `error contacting catalog service` or similar.

Reply:
My comment pertains to `/search` not `/show`.  My misunderstanding.